agent/systemd/notify.go

```diff
@@ -0,0 +1,33 @@
+package systemd
+
+import (
+	"errors"
+	"net"
+	"os"
+)
+
+var NotifyNoSocket = errors.New("No socket")
+
+// Notifier provides a method to send a message to systemd.
+type Notifier struct{}
+
+// Notify sends a message to the init daemon. It is common to ignore the error.
+func (n *Notifier) Notify(state string) error {
+	addr := &net.UnixAddr{
+		Name: os.Getenv("NOTIFY_SOCKET"),
```

Comment:
Should this be fleshed out into a normal config value that's also settable via CLI arg/config file?